jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java

```diff
@@ -579,20 +580,27 @@ public String toString()
      * when calling {@link Response#write(boolean, ByteBuffer, Callback)} to wrap the passed in {@link Callback}
      * so that the {@link RetainableByteBuffer} used can be released.
      */
-    private static class WriteErrorCallback extends Callback.Nested
+    private static class WriteErrorCallback implements Callback
     {
-        private final Retainable _retainable;
+        private final Callback _callback;
+        private final RetainableByteBuffer _buffer;
 
-        public WriteErrorCallback(Callback callback, Retainable retainable)
+        public WriteErrorCallback(Callback callback, RetainableByteBuffer retainable)
         {
-            super(callback);
-            _retainable = retainable;
+            _callback = callback;
+            _buffer = retainable;
         }
 
         @Override
-        public void completed()
+        public void succeeded()
         {
-            _retainable.release();
+            ExceptionUtil.callAndThen(_buffer::release, _callback::succeeded);
+        }
+
+        @Override
+        public void failed(Throwable x)
+        {
+            ExceptionUtil.callAndThen(x, t -> _buffer.remove(), _callback::failed);
```

Comment:
So in the case of an async abort, `failed` would be called, removing the buffer and decrementing its ref counter then `succeeded` would be called to, trying to decrement the ref counter once more.